Remove Quirk triggerOptionalStorageAccessQuirk for domains not part of Kinja anymore

karlcow opened the pull request:
#### 17d716bf82153f7018b0c8fea755d4825ef0a9ff
<pre>
Remove Quirk triggerOptionalStorageAccessQuirk for domains not part of Kinja anymore
<a href="https://bugs.webkit.org/show_bug.cgi?id=279581">https://bugs.webkit.org/show_bug.cgi?id=279581</a>
<a href="https://rdar.apple.com/135854408">rdar://135854408</a>

Reviewed by Brent Fulgham.

kinjaQuirks defines a list of domains to be used by the Quirk
triggerOptionalStorageAccessQuirk. This list is then used later in
addition to isKinjaLoginAvatarElement (another Quirk)

if (kinjaQuirks.get().contains(domain)
    &amp;&amp; isKinjaLoginAvatarElement(element)) { … }

Some of these domains are not part anymore of Kinja Login system. They
can be removed.

The list of domains to remove is:

* avclub.com
* deadspin.com
* jezebel.com
* lifehacker.com
* thetakeout.com
* theonion.com

It would be good to check for the remaining domains if the quirk is
needed at all. The markup defined in isKinjaLoginAvatarElement still
exists.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):

Canonical link: <a href="https://commits.webkit.org/283581@main">https://commits.webkit.org/283581@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d1c873e18876c57af1e06b1434f18d3c4ec5c6f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66603 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45978 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19225 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70637 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/17736 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/68721 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53777 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17592 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53370 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11959 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69670 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42331 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57634 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34036 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39002 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15025 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16090 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60913 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15366 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72339 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10560 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14808 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60697 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10592 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57705 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61026 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14724 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8687 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2308 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41785 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42862 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44045 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42605 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
LGTM. I assume all of these have been validated to work without the Quirk?

karlcow committed:
Remove Quirk triggerOptionalStorageAccessQuirk for domains not part of Kinja anymore
https://bugs.webkit.org/show_bug.cgi?id=279581
rdar://135854408

Reviewed by Brent Fulgham.

kinjaQuirks defines a list of domains to be used by the Quirk
triggerOptionalStorageAccessQuirk. This list is then used later in
addition to isKinjaLoginAvatarElement (another Quirk)

if (kinjaQuirks.get().contains(domain)
    && isKinjaLoginAvatarElement(element)) { … }

Some of these domains are not part anymore of Kinja Login system. They
can be removed.

The list of domains to remove is:

* avclub.com
* deadspin.com
* jezebel.com
* lifehacker.com
* thetakeout.com
* theonion.com

It would be good to check for the remaining domains if the quirk is
needed at all. The markup defined in isKinjaLoginAvatarElement still
exists.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):

Canonical link: https://commits.webkit.org/283581@main
